src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -2350,6 +2355,34 @@ mongoc_uri_get_ssl (const mongoc_uri_t *uri) /* IN */
    return mongoc_uri_get_tls (uri);
 }
 
+const char *
+mongoc_uri_get_server_monitoring_mode (const mongoc_uri_t *uri)
+{
+   BSON_ASSERT (uri);
+
+   return mongoc_uri_get_option_as_utf8 (uri, MONGOC_URI_SERVERMONITORINGMODE, "auto");
+}
+
+
+bool
+mongoc_uri_set_server_monitoring_mode (mongoc_uri_t *uri, const char *value)
+{
+   BSON_ASSERT (uri);
+   BSON_ASSERT (value);
+
+   if (!bson_utf8_validate (value, strlen (value), false)) {
+      return false;
+   }
```

Comment:
```suggestion
```

Suggest removing extra validation check. The `strcmp` checks below restrict the accepted inputs to exactly one of those three.